pkg/controller/admissionchecks/multikueue/admissioncheck.go

```diff
@@ -308,49 +218,69 @@ func queueReconcileForConfigUsers(ctx context.Context, config string, c client.C
 	return nil
 }
 
-type secretHandler struct {
+type mkClusterHandler struct {
 	client client.Client
 }
 
-var _ handler.EventHandler = (*secretHandler)(nil)
+var _ handler.EventHandler = (*mkClusterHandler)(nil)
 
-func (s *secretHandler) Create(ctx context.Context, event event.CreateEvent, q workqueue.RateLimitingInterface) {
-	if err := s.queue(ctx, event.Object, q); err != nil {
-		ctrl.LoggerFrom(ctx).V(5).Error(err, "Failure on create event", "secret", klog.KObj(event.Object))
+func (m *mkClusterHandler) Create(ctx context.Context, event event.CreateEvent, q workqueue.RateLimitingInterface) {
+	mkc, isMKC := event.Object.(*kueuealpha.MultiKueueConfig)
```

Comment:
This might be a sign that we are missing an integration test for the case where a MKCluster is added after the MKConfig is created.

Reply:
> This might be a sign that we are missing an integration test for the case where a MKCluster is added after the MKConfig is created.

Yes, that's right. That is what I wanted to say here: `Also, can you verify if this handler fires reconcile when the MultiKueueCluster is created in tests?`

Reply:
added a test